fs/fs.go

```diff
@@ -417,31 +417,18 @@ func (fs *filesystem) MountLocal(ctx context.Context, mountpoint string, labels
 	if err != nil {
 		return fmt.Errorf("cannot parse image ref (%s): %w", imageRef, err)
 	}
-	remoteStore, err := newRemoteStore(refspec, client)
+	repo, err := newRemoteStore(refspec, client)
 	if err != nil {
 		return fmt.Errorf("cannot create remote store: %w", err)
 	}
+	remoteStore := newRemoteBlobStore(repo)
```

Comment:
This looks redundant and we are not using `repo` except feeding it to `newRemoteBlobStore`. Probably we can combine the two into the new function.